### ISSUE TITLE
tag expansion doesn't work inside front matter `tags:`

### ISSUE THREAD

**Issue by jimgraham** [foambubble/foam#1184]
### Describe the bug

Tag expansion works when a `#` character is detected while typing. There are improvements suggested in #1182 and #1183 to restrict this outside of markdown headers.

One place the expansion should work _without_ looking for a `#` character is in the front-matter

```markdown
---
tags: tag1 tag2
---
```

If you use a `#`, it is interpreted as a comment

```markdown
---
tags: #tag1 #tag2
---
```

![image](https://user-images.githubusercontent.com/430293/230419259-a1ead787-6c57-41ec-a022-52604119da4e.png)

The tag expansion, however, _is_ triggered

![image](https://user-images.githubusercontent.com/430293/230419520-03fc4999-c99c-4f95-a2bf-86f02a39e363.png)

### Steps to Reproduce the Bug or Issue

1. create a document with two tags `first-tag` , `second-tag`
2. in a separate document, add front matter with 
```markdown
---
tags: fir
```

### Expected behavior

In the example above, I expect the tag-expansion to show and to offer `first-tag` 

### Operating System Version

OSX 13.3

### Visual Studio Code Version

1.77

### Additional context

I'm happy to take this on like in #1183

**Comment by riccardoferretti** [comment]
Thanks for fixing the previous issue, and to offer to address this one to.
I preferred to keep the two separate, now that the PR has been merged feel free to move forward, and thanks again!

**Comment by jimgraham** [comment]
closed by #1191
